src/parser/esprima_cpp/esprima.cpp

```diff
@@ -440,10 +458,10 @@ class Parser : public gc {
         }
 
         String* value;
+        value = new ASCIIString("ILLEGAL");
```

Comment:
Newly allocating an ASCIIString for the `value` by default may be inefficient because it can be re-allocated later. Since each String object is dynamically managed by GC, it would be better to allocate each String object when it is really necessary.

Reply:
Thanks for the notice, i forgot to consider this. I'm gonna update this soon.